iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/ArmParser.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.parser;
+
+import java.util.List;
+import javax.annotation.Nullable;
+import org.sonar.iac.arm.tree.api.ArmTree;
+import org.sonar.iac.arm.tree.impl.json.FileImpl;
+import org.sonar.iac.common.extension.ParseException;
+import org.sonar.iac.common.extension.TreeParser;
+import org.sonar.iac.common.extension.visitors.InputFileContext;
+import org.sonar.iac.common.yaml.YamlParser;
+import org.sonar.iac.common.yaml.tree.FileTree;
+
+public class ArmParser implements TreeParser<ArmTree> {
```

Comment:
I see you are later using a `YamlParser`.
I checked how `CloudFormationParser` is done, it is currently extending directly from the `YamlParser`.
Do you think it would make sense here? Seems a bit strange to instantiate a different parser in your parser.

Reply:
The `CloudFormationParser` extends `YamlParser` to use `CloudformationConverter` to do some magic, we don't need it.
The second reason is that methods defined in `YamlParser` return types from `org.sonar.iac.common.yaml.tree` package and we would have another tree structure just for ARM.
Finally, it will be easier to extend it or refactor it in case of implementing Bicep parser in the future.